src/core/mac/sub_mac.cpp

```diff
@@ -254,43 +253,28 @@ Error SubMac::Receive(uint8_t aChannel)
 }
 
 #if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
-Error SubMac::CslSample(uint8_t aPanChannel)
+void SubMac::CslSample(void)
 {
     Error error = kErrorNone;
 
-    if (!IsCslChannelSpecified())
-    {
-        mCslChannel = aPanChannel;
-    }
-
 #if OPENTHREAD_CONFIG_MAC_FILTER_ENABLE
     VerifyOrExit(!mRadioFilterEnabled, error = Get<Radio>().Sleep());
 #endif
 
-    switch (mCslState)
-    {
-    case kCslSample:
-        error = Get<Radio>().Receive(mCslChannel);
-        break;
-    case kCslSleep:
+    SetState(kStateCslSample);
+
 #if !OPENTHREAD_CONFIG_MAC_CSL_DEBUG_ENABLE
-        error = Get<Radio>().Sleep(); // Don't actually sleep for debugging
+    IgnoreError(Get<Radio>().Sleep()); // Don't actually sleep for debugging
```

Comment:
From https://github.com/openthread/openthread/pull/7783#discussion_r911541071. 
> Actually I don't think it was correct in the previous implementation that we switch to receive somehow asynchronously just after finishing another MAC operation. We wouldn't be considering the calculated `SubMac::GetCslWindowEdges` and probably be receiving not long enough to receive a full frame.
> 
> Instead, I believe it's better to always sleep here (this comes from the shouldSleep decision in `Mac::UpdateIdleMode()`) and just let the timer trigger for a proper full reception slot.

I still think it may be safer and better to enter rx if we are within the sample window (even if it is middle of it).

It may be early part of the sample window and/or radio may have been in rx mode already before this (e.g., call to `CslSample()` happened while we were in rx before).

If we call `Sleep()` then we, for sure, cannot receive anything in this window.

Reply:
Well, I was just trying to save some code size, but it's true that in some cases it could be useful to start receiving.